monkey/tests/unit_tests/infection_monkey/network/test_network_tools.py

```diff
@@ -0,0 +1,13 @@
+import pytest
+
+from infection_monkey.network.tools import connect
+
+
+def test_connect_raises_with_empty_list():
+    with pytest.raises(ConnectionError):
+        connect([])
+
+
+def test_connect_raises_with_bad_data():
+    with pytest.raises(ValueError):
+        connect(["no-port"])
```

Comment:
Probably one more test where we check if `socket.create_connection` is called with right arguments.

Reply:
I don't know a good unobtrusive way to do this without modifying the method signature